soccer/gameplay/plays/defense/defend_goal.py

```diff
@@ -0,0 +1,67 @@
+import main
+import robocup
+import behavior
+import constants
+import enum
+
+import standard_play
+import tactics.positions.submissive_goalie as submissive_goalie
+import tactics.positions.submissive_defender as submissive_defender
+import evaluation.opponent as eval_opp
+import tactics.positions.wing_defender as wing_defender
+import skills.mark as mark
+import tactics.defense
+
+
+## Play that uses submissive defenders and wingers to defend
+#  an attack close to our goal.
+#  
+#  By default, we will use standard defense (two submissive 
+#  defenders, one goalie) and two wing defense robots. The
+#  remaining robot will mark the highest threat robot. 
+class DefendGoal(standard_play.StandardPlay):
+    def __init__(self, num_defenders=3, num_wingers=2):
+        super().__init__(continuous=True)
+
+        self.num_defenders = num_defenders
+        self.num_wingers = num_wingers
+
+        self.add_transition(behavior.Behavior.State.start,
+                            behavior.Behavior.State.running, lambda: True,
+                            'Immediately')
+
+        # Use standard defense
+        self.add_subbehavior(
+            tactics.defense.Defense(), 'defense', required=False)
+
+        self.add_subbehavior(mark.Mark(), 'mark')
+
+        # Keep track of which robots are currently being defended
+        self.defended = {}
+        for i in range(len(main.their_robots())):
+            self.defended[i] = False
+
+        for i in range(self.num_wingers):
+            self.add_subbehavior(wing_defender.WingDefender(),
+                                 'winger' + str(i))
+
+    def execute_running(self):
+        for i in range(self.num_wingers):
+            bhvr = self.subbehavior_with_name('winger' + str(i))
```

Comment:
Does this make a list?

Reply:
No, this tells the wing defenders which robots to mark